generator/src/main/java/com/scottlogic/deg/generator/generation/fieldvaluesources/CombiningFieldValueSource.java

```diff
@@ -30,10 +30,12 @@
 import static com.scottlogic.deg.generator.utils.SetUtils.stream;
 
 public class CombiningFieldValueSource implements FieldValueSource {
-    private final List<FieldValueSource> underlyingSources;
+    private final List<FieldValueSource<Object>> underlyingSources;
 
     public CombiningFieldValueSource(List<FieldValueSource> underlyingSources) {
-        this.underlyingSources = underlyingSources;
+        this.underlyingSources = underlyingSources.stream()
```

Comment:
Would it be worth making the parameter `underlyingSources` a parameterized type instead of casting in this method?

(where parameter is the declaration in the function signature, and the parameterized type is `List<FieldValueSource<Object>>`... it's annoying when terminology clashes)

Reply:
it's only used for adding null.

I'm going to try and rework it as a nullAppendingValueSource which is generically typed